File: 1803-average-waiting-time/1803-average-waiting-time.py
class Solution:
    def averageWaitingTime(self, customers: List[List[int]]) -> float:
        
        t = customers[0][0] + customers[0][1]
        a = customers[0][1]

        for x in customers[1:]:
            t = max(t, x[0]) + x[1]
            a += t - x[0]

        print(a)
        return a/len(customers)
            

File: 1803-average-waiting-time/1803-average-waiting-time.cs
public class Solution {
    public double AverageWaitingTime(int[][] customers) {
        
        double sum =0;
        double finalSum = 0;
        double av =0;
        for(int i=0;i<customers.Length;i++){
            if(i==0 || sum < customers[i][0])
                {
                    sum = customers[i][1]+customers[i][0];
                    finalSum = sum - customers[i][0];
                }
           else{
           sum=sum+customers[i][1];  
           finalSum= Math.Abs(sum - customers[i][0]);
           }
           av+=finalSum;
           Console.WriteLine(finalSum);
        }
        return av/customers.Length;
    }
}